### ISSUE TITLE
Feature: Add Progress bars

### ISSUE THREAD

**Issue by brandyscarney** [ionic-team/ionic#6669]
https://www.google.com/design/spec/components/progress-activity.html

http://g.recordit.co/NEQuvntLcF.gif

**Comment by ghenry22** [comment]
Progress bars as a standard component would be really nice, I would definitely make use of this if the feature was added.

**Comment by lordgreg** [comment]
+1

**Comment by movrack** [comment]
+1

**Comment by movrack** [comment]
Finally for our case we will use progress of html5. We just have to design  it for each platform

**Comment by jgw96** [comment]
Hello! We are moving our feature requests to a new feature request [doc](https://docs.google.com/document/d/1vZ339ibia2DA9JVSfO5eGSvej868vooex61xJ5Za8YM/edit). I have moved this feature request to the doc and because of this I will be closing this issue for now. Thanks for using Ionic!

**Comment by Bouzmine** [comment]
It would be nice to have an option to integrate it with the navbar, seen in Safari iOS, for example.

**Comment by rafaelwdantas** [comment]
+1

**Comment by ghenry22** [comment]
https://github.com/valor-software/ng2-bootstrap

this can be easily included into your ionic2 project and it gives you some decent consistent progress bar styling across platforms.  Still a native ionic component would be nice but this definitely works, I have rolled it into my own project so that I could have some nice progress bars for downloads.

**Comment by aspidvip** [comment]
+1

**Comment by sburak** [comment]
+1

**Comment by jgw96** [comment]
Hello all! While this is an awesome feature request it is not something that we plan on doing anytime soon. Because of this I am going to move this to our internal feature tracking repo for now as it is just "collecting dust" here. Once we decide to implement this I will move it back. Thanks everyone for using Ionic!

**Comment by jgw96** [comment]
This issue was moved to driftyco/ionic-feature-requests#38

**Comment by danielabbatt** [comment]
This is something I've been looking around at for a while and looking for a good implementation, out of the box would be amazing.

**Comment by slzr** [comment]
meanwhile we waiting for the `ionic progress bar component`, I leave this [simple pen](http://codepen.io/jsalazart/pen/aJdmvw?editors=1100) with a material design looks, inspired in this [css-tricks tutorial](https://css-tricks.com/html5-progress-element/) and the [materialize css indeterminate progress bar](http://materializecss.com/preloader.html)
```html
<progress max="100" value="50"></progress>
```
```sass
.progress, progress[value] {
  -webkit-appearance: none;
  appearance: none;
  border: none;
  display:block;
  height: 10px;
  &::-webkit-progress-bar {
    background-color: lighten( map-get($colors, primary), 35% );
  }
  &::-webkit-progress-value {
    background-color: map-get($colors, primary);
  }
}
```
in the next days I'll try to make it a sass mixin to improve the use

**Comment by realappie** [comment]
@brandyscarney I am willing to create a progress bar for android, just want a heads up whether anyone on the team is already working on one before starting!

**Comment by brandyscarney** [comment]
@realappie It is not currently being worked on. I started progress on it awhile ago but it is so outdated now that it would probably be easier to start from scratch. 🙂

**Comment by realappie** [comment]
@brandyscarney beside the progress bar having four modes, are there any other requirements? Also the progress bar in the gif is a material design one, are we going to be re-using that for iOS and windows as well or are they supposed to get a different looking loading indicator.

**Comment by realappie** [comment]
@jgw96 Would you mind re-opening this as your link is invalid ? :(

**Comment by brandyscarney** [comment]
@realappie I would say the progress bars should follow the Material Design requirements: https://material.io/guidelines/components/progress-activity.html

The first iteration doesn't have to do _everything_, but eventually we want to get there. I think for now we can make them all look the same and then tweak them for iOS / Windows later. The Material Design team has some good documents to refer to: https://docs.google.com/document/d/1lBJkORFeB7y_RMRtXfT4bIdz9PnUYLn0HYtSClayXt4/preview

I was thinking we could have one component `ion-progress` and then have the following inputs:

```
type: 'bar' | 'circle' = 'bar';
indicator: 'determinate' | 'indeterminate' | 'buffer' | 'query' = 'determinate';
```

Since we already use `mode` for our theming. I'm open to ideas though. 🙂

**Comment by realappie** [comment]
@brandyscarney No that sounds like a great idea! Also don't we already have an `ion-spinner` which would be the same as the progress of type circular?

I'm currently busy with the determinate version of the progress bar, the rest will be easy to implement but the animation of this one is tricky !

**Comment by realappie** [comment]
@brandyscarney I finished the progress bars, but I'm getting some linting errors because I'm using the property `left` is prohibited. Should I refactor my code or would my code be accepted?

![](https://i.gyazo.com/0331192939f8a939074674d1fb4fe865.gif)

**Comment by realappie** [comment]
@brandyscarney How do the linting rules actually work? Because the output of the linter is somewhat verbose and I want to understand it so I can contribute to ionic in the future as well.

If you would rather fix it, I will submit a PR :D

**Comment by brandyscarney** [comment]
@realappie Each linter rule and a description of what they do can be found here: https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md

For right to left support we have just disabled certain properties from being used via `PropertySpelling` in favor of the mixins. We could probably enhance the linter to provide more information, but this is why it's happening.

**Comment by realappie** [comment]
@brandyscarney Ah I see, thank you! I will give RTL support a shot but if it takes me too long I will just submit the PR considering I have never done anything with RTL.

**Comment by brandyscarney** [comment]
Sounds good, feel free to submit the PR and either @AmitMY or myself can help out. :smile:

**Comment by realappie** [comment]
@brandyscarney Maybe change the v2 label to v4? 👍

**Comment by phpfs** [comment]
Any updates to this? :)

**Comment by harshabyloor** [comment]
@brandyscarney @realappie  Is there working indeterminate progress bar? Can you please share details
